### PR TITLE
Close tab

### LINKED ISSUE #71: Advanced features for tabs
Implement features for Tabs
- [x] Open a tab
    - [x] CLI
    - [x] GUI
- [x] Closing of tabs
    - [x] CLI
    - [x] GUI
- [x] Save state of GUI in storage

### PR DESCRIPTION
**Pull request by tiuweehan**
Closes #71

Additional Details:
- Refactor ID to be a separate field

### CODE REVIEW

**Comment by tiuweehan** [comment]
These should be all my features for v1.3, without the tests and documentation

File: src/test/java/seedu/algobase/logic/commands/CloseTabCommandTest.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+class CloseTabCommandTest {
+
+    @Test
+    public void constructor_nullProblem_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new CloseTabCommand(null));
+    }
+
+    /**
+     * A Model stub that always accepts the problem being added.
```
Comment (jiayushe) [inline src/test/java/seedu/algobase/logic/commands/CloseTabCommandTest.java]: Update this line.

File: src/test/java/seedu/algobase/logic/commands/OpenTabCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        );
+
+        assertEquals(
+            String.format(OpenTabCommand.MESSAGE_SWITCH_SUCCESS, index1.getOneBased()),
+            commandResult1Copy.getFeedbackToUser()
+        );
+    }
+
+    /**
+     * A Model stub that always accepts the problem being added.
```
Comment (jiayushe) [inline src/test/java/seedu/algobase/logic/commands/OpenTabCommandTest.java]: Same as above.

File: src/main/java/seedu/algobase/ui/PlanCard.java [hunk 3]
```diff
@@ -1,18 +1,30 @@
...
 
-
 /**
  * An UI component that displays information of a {@code Plan}.
  */
 public class PlanCard extends UiPart<Region> {
 
+    private static final Logger logger = LogsCenter.getLogger(PlanCard.class);
+
+
```
Comment (jiayushe) [inline src/main/java/seedu/algobase/ui/PlanCard.java]: Remove one line.

File: src/main/java/seedu/algobase/model/Id.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+    private Id(String idValue) {
+        this.idValue = idValue;
+    }
+
+    /**
+     * Generate Id based on unix timestamp.
+     */
+    public static Id generateId() {
+        return new Id(UUID.randomUUID().toString());
```
Comment (jiayushe) [inline src/main/java/seedu/algobase/model/Id.java:19]: Does UUID enforce uniqueness over time? Update the header by the way.

**Review by jiayushe — APPROVED** [review]
I didn't do functional testing. Otherwise LGTM. Just do the minor modifications as commented.

File: src/main/java/seedu/algobase/model/Id.java (hunk 4 above)
Comment (tiuweehan) [reply src/main/java/seedu/algobase/model/Id.java:19]: Good enough https://stackoverflow.com/questions/20999792/does-randomuuid-give-a-unique-id

**Review by le0tan — APPROVED** [review]
LGTM. Please add UG and update DG for this feature in the next PR =)

**Pull request merged**